rfcs/ALL/Access Control Rules.md

```diff
@@ -0,0 +1,134 @@
+
+The following table demonstrates how the matching will work on a key-expression(KE) in request and in the rules set:
+
+| KE in request | KE in rules set | Match |
+|---------------|---------------|-------|
+| test/demo/a         | test/demo/a         | yes   |
+| test/demo/a         | test/demo/*         | yes   |
+| test/\*/\*         | test/**          | yes   |
+| test/demo/a         | **            | yes   |
+| test/demo/a         | test/*/a         | yes   |
+| test/demo/*         | test/*/a         | no    |
+| test/**          | test/demo/a         | no    |
+
+The same behaviour extends for our DSL support as well:
+
+| KE in request | KE in rules set | Match |
+|---------------|---------------|-------|
+| test/demo/a         | test/d$*/a         | yes   |
+| test/demo/a         | t$*/**            | yes   |
+| test/d$*/a        | t$*/**            | yes   |
+| test/demo/*         | test/d$*/a         | no    |
+| test/d$*          | test/demo/a         | no    |
+
+For verbatims, the subpart of the key-expression starting with `@` has to be *exactly* the same in the rules set for a match. The other parts of the key-expression follow matching rules as before.
+
+| KE in request | KE in rules set | Match |
+|---------------|---------------|-------|
+| test/@demo/a         | test/demo/a         | no   |
+| test/demo/a         | test/@demo/a         | no   |
+| test/@demo/a         | test/@demo/*         | yes   |
+| test/@demo/a         | **            | no   |
+| test/@demo/a         | test/*/a         | no   |
+| test/@demo/*         | */@demo/a         | no    |
+
+If the match happens then the result will be as set in the explicit rules. If not, then the default permission will take over. For example, if the default permission is `allow` and the key-expressions in the *explicit deny* rules are any of the key-expressions like `test/demo/a` , `test/demo/*` or `test/demo/**` then a request on `test/demo/a` will match with the *explicit deny* KeTree and will be denied. However, given the same rules set, a request on `test/**` will not match (since it is a superset) and therefore the request will be allowed to go through. Therefore, extra care needs to be taken while devising the rules, and especially so when using wildcards.
+
+
+## Performance
+
+Given, zenoh's priority is performance, a lot of care was taken while adding access control features to the codebase, to keep the performance as high as possible. However, as with any other piece of software, security comes with a price in terms of performance. However, having done multiple tests, we can share some tips to improve performance:
```

Comment:
Nitpicking:
- Remove comma after "Given"? Idk it doesn't sound correct to me 😅
- Capitalize "Zenoh"